src/deluge/gui/views/audio_clip_view.cpp

```diff
@@ -439,184 +471,141 @@ ActionResult AudioClipView::buttonAction(deluge::hid::Button b, bool on, bool in
 							changeUnderlyingSampleLength(clip, sample, newLength, oldLength, oldLengthSamples);
-
-							goto needRendering;
+							uiNeedsRendering(this, 0xFFFFFFFF, 0);
+						}
+					}
+				}
+				// =========== Handling START marker =============
+				else if (startMarkerVisible) {
+					if (x == startSquareDisplay || x == endSquareDisplay) {
+						startMarkerVisible = false; // Toggle start marker off
+						uiTimerManager.unsetTimer(TimerName::UI_SPECIFIC);
+						uiNeedsRendering(this, 0xFFFFFFFF, 0);
+					}
+					else {
+						Sample* sample = getSample();
+						if (sample) {
+							int32_t newStartTicks =
+							    x * currentSong->xZoom[NAVIGATION_CLIP] + currentSong->xScroll[NAVIGATION_CLIP];
+							int32_t oldLength = clip->loopLength;
+							uint64_t oldLengthSamples = clip->sampleHolder.getDurationInSamples(true);
+							changeUnderlyingSampleStart(clip, sample, newStartTicks, oldLength, oldLengthSamples);
+							uiNeedsRendering(this, 0xFFFFFFFF, 0);
 						}
 					}
 				}
-
-				// Or, marker not already visible
 				else {
+					// No marker is visible. Are we near the end or start?
 					if (x == endSquareDisplay || x == endSquareDisplay + 1) {
 						endMarkerVisible = true;
-needRendering:
-						uiTimerManager.setTimer(TimerName::UI_SPECIFIC, kSampleMarkerBlinkTime);
+						startMarkerVisible = false;
 						blinkOn = true;
+						uiTimerManager.setTimer(TimerName::UI_SPECIFIC, kSampleMarkerBlinkTime);
 						uiNeedsRendering(this, 0xFFFFFFFF, 0);
 					}
+					else if (x == startSquareDisplay) {
+
+						// WIP: Allow the user to trim from the start of the audio clip
+						if (runtimeFeatureSettings.get(RuntimeFeatureSettingType::TrimFromStartOfAudioClip)) {
+							startMarkerVisible = true;
+							endMarkerVisible = false;
+							blinkOn = true;
+							uiTimerManager.setTimer(TimerName::UI_SPECIFIC, kSampleMarkerBlinkTime);
+							uiNeedsRendering(this, 0xFFFFFFFF, 0);
+						}
+					}
 				}
 			}
 		}
 	}
 	else if (x == kDisplayWidth) {
-		if (isUIModeActive(UI_MODE_HOLDING_SONG_BUTTON)) {
-			if (sdRoutineLock) {
-				return ActionResult::REMIND_ME_OUTSIDE_CARD_ROUTINE;
-			}
-			if (!on) {
-				view.activateMacro(y);
-			}
-			return ActionResult::DEALT_WITH;
-		}
+		// side column actions or macros, if any
 	}
-
 	return ActionResult::DEALT_WITH;
 }
+
+// ----------- "End" pointer logic -----------
 void AudioClipView::changeUnderlyingSampleLength(AudioClip* clip, const Sample* sample, int32_t newLength,
                                                  int32_t oldLength, uint64_t oldLengthSamples) const {
 	uint64_t* valueToChange;
 	int64_t newEndPosSamples;
 
 	uint64_t newLengthSamples =
-	    (uint64_t)(oldLengthSamples * newLength + (oldLength >> 1)) / (uint32_t)oldLength; // Rounded
-	// AudioClip reversed
-	if (clip->sampleControls.reversed) {
+	    (uint64_t)(oldLengthSamples * newLength + (oldLength >> 1)) / (uint32_t)oldLength; // rounding
```

Comment:
please use C++ style casting and division over C casting and shifting
```suggestion
	    static_cast<uint64_t>(oldLengthSamples * newLength + (oldLength / 2)) / static_cast<uint32_t>(oldLength); // rounding
```